openmdao/components/tests/test_balance_comp.py

```diff
@@ -97,6 +98,51 @@ def test_create_on_init(self):
         for (of, wrt) in cpd['balance']:
             assert_almost_equal(cpd['balance'][of, wrt]['abs error'], 0.0, decimal=5)
 
+    def test_balance_comp_options_exclude_no_error(self):
+
+        prob = om.Problem()
+
+        bal = om.BalanceComp()
+        bal.add_balance('x', val=1.0)
+
+        prob.model.add_subsystem(name='balance', subsys=bal)
+
+        recorder = om.SqliteRecorder('cases.sql')
+
+        prob.model.add_recorder(recorder)
+
+        prob.model.recording_options['record_inputs'] = True
+        prob.model.recording_options['record_outputs'] = True
+        prob.model.recording_options['record_residuals'] = True
+        bal.recording_options['record_metadata'] = False
+
+        prob.setup()
+
+        msg = ("Trying to record options which cannot be pickled on system "
```

Comment:
This warning text in msg doesn't match the warning text that would be printed, so I think this test would pass even without the fix. I recommend temporarily setting the `recordable` to True in that balance option, and fix this so that it fails, then restore it to False.

Reply:
fixed